### DISCUSSION TITLE
Make the brush able to draw areas

### DISCUSSION THREAD

**Discussion by TORD31** [Mojang/minecraft-editor#94] [Feedback]
When you are building for example a village in editor mode and you want to connect your buildings with paths with the brush tool you will have a problem if you want a path that is wider than only one block. When you change the brush size to something bigger than one, you always draw cubes which means that you would also replace the blocks under the actual path you wanted do draw.

To make something like this easier it would be useful to add an "draw area" option to the brush which would make the brush only replace the top blocks of the region you clicked on and not also the blocks under them.

**Comment by minecraftchloe** [comment]
This is a GREAT idea, and something that is a prime candidate (and that we've seen prototyped internally in testing the Editor API) for a "Path Generator" extension! Stay tuned for Editor v0.4 release coming soon.

**Comment by minecraftchloe** [comment]
1. I think this is a great candidate for an actual core feature within the brush editor experience -- drawing paths between areas. 
2. A workaround could be to use Selection --> Freeform mode + Fill. That doesn't necessarily adapt to the terrain, but on flat surfaces, it'll just replace the top blocks of the region you click on. 
3. If you wanted to try your hand @ creating your own path extension, this should be possible with the release of the Editor API #130 -- we did some internal demos, and one of our team members was able to create a rough example! If you try making this, let us know what suggestions you have for how we can expand the API to create what your ideal is! 
 
<img width="541" alt="Screenshot 2023-06-29 at 10 15 19 AM" src="https://github.com/Mojang/minecraft-editor/assets/56832421/0ea67081-831e-4175-bc94-a1ee45e6effd">
